api/envoy/extensions/filters/listener/tls_inspector/v3/tls_inspector.proto

```diff
@@ -25,6 +25,11 @@ message TlsInspector {
   // Populate ``JA3`` fingerprint hash using data from the TLS Client Hello packet. Default is false.
   google.protobuf.BoolValue enable_ja3_fingerprinting = 1;
 
+  // Populate ``JA4`` fingerprint hash using data from the TLS Client Hello packet.
+  // ``JA4`` is an improved version of ``JA3`` that includes TLS version, ciphers, extensions,
+  // and ALPN information in a hex format. Default is false.
```

Comment:
Can you mention the order of precedence if ja4 and ja3 are both enabled?

Reply:
It will compute and populate both JA3 and JA4 which is technically correct as someone might need to record both the JA3 as well as JA4 fingerprints.

Reply:
got it thanks for clarifying!